lib/OpenQA/WebAPI/Plugin/Helpers.pm

```diff
@@ -480,6 +481,15 @@ sub _compose_job_overview_search_args ($c) {
     # allow filtering by group ID or group name
     $search_args{groupids} = [map { $_->id } @groups] if @groups;
 
+    # allow excluding jobs by group ID (0 to exclude groupless jobs)
+    if ($v->is_valid('not_groupid')) {
+        my @not_group_ids = @{$v->every_param('not_groupid')};
+        $search_args{groupid} = {-not_in => \@not_group_ids};
+        if (grep { $_ == 0 } @not_group_ids) {
+            $search_args{groupid} = {-not_in => \@not_group_ids, '!=' => undef};
```

Comment:
This is just overriding the previous assignment. I'm not sure whether this is going to work as intended but we'll see when you've added tests.